### PR TITLE
Implement `initiatorHandshake` method

### PR DESCRIPTION
**Pull request by EmilIvanichkovv**
The PR implements the functionality required in this issue https://github.com/status-im/status-node-manager/issues/7

### CODE REVIEW

File: libs/waku-utils/waku_handshake_utils.nim [hunk 1]
```diff
@@ -165,46 +142,82 @@ proc prepareHandShakeMsg*(rng: ref HmacDrbgContext,
...
+    initiatorHS: HandshakeState = initHS(initiatorInfo, qr, true)
+
+    initiatorHSResult: HandshakeResult
+
+  let initialMessage =
+    prepareHandShakeInitiatorMsg(rng, contentTopic, initiatorInfo,
+        qrMessageNameTag, initiatorMessageNametag, initiatorHS, initiatorStep)
+
+  proc initiatorHandshakeHandler(topic: PubsubTopic, msg: WakuMessage): Future[
+      void] {.async, gcsafe.} =
```
Comment (Ivansete-status) [inline libs/waku-utils/waku_handshake_utils.nim:193]: Simple nitpick: the `async` pragma statement is equivalent to `gcsafe, raises: []`
```suggestion
      void] {.async.} =
```

File: libs/waku-utils/waku_handshake_utils.nim [hunk 2]
```diff
@@ -165,46 +142,82 @@ proc prepareHandShakeMsg*(rng: ref HmacDrbgContext,
...
+    if msg.contentTopic == contentTopic:
+      let readPayloadV2 = decodePayloadV2(msg).get()
+      if readPayloadV2.messageNametag == initiatorMessageNametag:
+        handleHandShakeMsg(rng, pubSubTopic, contentTopic, step = 2,
```
Comment (Ivansete-status) [inline libs/waku-utils/waku_handshake_utils.nim:197]: What does `step` mean in this context?

**Review by Ivansete-status — COMMENTED** [review]
Thanks for the PR!
Some comments so far:
1. If possible, I would avoid changing the format. I prrefer vertical style.
2. I've added some comments re the timings. If possible, I would avoid having such hard-coded waits and instead, have an active wait until certain state is met. With that approach, a timeout can be used.

File: libs/waku-utils/waku_handshake_utils.nim (hunk 2 above)
Comment (EmilIvanichkovv) [reply libs/waku-utils/waku_handshake_utils.nim:197]: `step` indicates which phase of the waku pairing we are on. But this is only used for the logs in `handleHandShakeMsg`

**Comment by EmilIvanichkovv** [comment]
Ty for the review @Ivansete-status 
> If possible, I would avoid changing the format. I prrefer vertical style.
My goal is at some point to introduce automatic forming in the repo. That is why I switched to this pattern.

**Pull request merged**